router/router.go

```diff
@@ -1849,3 +1860,66 @@ func (rt *HandleT) Resume() {
 
 	rt.paused = false
 }
+
+// Currently the retry logic has been implemented for only OAuth
+func (rt *HandleT) SendToTransformerProxyWithRetry(val integrations.PostParametersT,
+	destinationJob types.DestinationJobT, accessToken string, retryCount int) (statusCode int, response string) {
+	var respStatusCode int
+	var respBodyTemp string
+
+	respStatusCode, respBodyTemp = rt.transformer.Send(val, rt.destName, accessToken)
+	if retryCount >= 2 {
+		// Retrial termination condition
+		return respStatusCode, respBodyTemp
+	}
+	fmt.Printf("SendToProxy Response Code: %d\n", respStatusCode)
+	fmt.Printf("SendToProxy Response: %s\n", respBodyTemp)
+	var errOutput oauth.ErrorOutput
+	if err := json.Unmarshal([]byte(respBodyTemp), &errOutput); err != nil {
+		// If respBodyTemp comes out with a string, then this will occur
+		// TODO: Should we send any other error here ?
+		return http.StatusInternalServerError, err.Error()
+	}
+	workspaceId := destinationJob.JobMetadataArray[0].WorkspaceId
+	var stCd int = 0
```

Comment:
need not initialise with values, they would be defaulted to 0 and ""

Reply:
👍🏼